packages/astro/src/runtime/server/render/component.ts

```diff
@@ -80,7 +80,7 @@ async function renderFrameworkComponent(
 	_props: Record<string | number, any>,
 	slots: any = {}
 ): Promise<RenderInstance> {
-	if (!Component && !_props['client:only']) {
+	if (!Component && !_props['client:only'] && _props['client:only'] !== "") {
```

Comment:
Add `_props['client:only'] !== ""` so that the following logic can be entered

Reply:
If I understand correctly, we are making this condition more specific so that we do not throw an error here, and instead go on to guess the renderer.

We previously checked whether `props["client:only"]` was falsy (null, undefined, 0, ""), but really what we care about is whether or not its provided. Could `"client:only" in _props === false` be a better check?

Reply:
Yeah. You are right! Thank you!